packages/mongo-nav/src/MongoNav.tsx

```diff
@@ -101,16 +121,60 @@ export default function MongoNav({
   admin = false,
   constructOrganizationURL: constructOrganizationURLProp,
   constructProjectURL: constructProjectURLProp,
+  onError = () => {},
+  onSuccess = () => {},
 }: MongoNavInterface) {
-  let data;
+  const [data, setData] = React.useState<DataInterface | undefined>(undefined);
+
+  const accountHost = hosts?.account ?? `https://account.mongodb.com`;
+  const cloudHost = hosts?.cloud ?? `https://cloud.mongodb.com`;
+  const universityHost = hosts?.university ?? `https://university.mongodb.com`;
+  const supportHost = hosts?.support ?? `https://support.mongodb.com`;
 
-  if (mode === Mode.Dev) {
-    data = devModeData;
+  function getProductionData() {
+    fetch(`${cloudHost}/user/shared`, {
+      credentials: 'include',
+      method: 'GET',
+    })
+      .then(response => response.json())
+      .then(data => {
+        setData(data);
+        onSuccess?.(data);
+      })
+      .catch(error => {
+        onError?.(error);
+        console.warn(error);
+      });
   }
 
-  if (!data) {
-    // Eventually this logic will be more robust, but for an alpha version will return null without data
-    return null;
+  function getFixtureData() {
```

Comment:
This was mostly for the sake of testing our handling without making actual fetch requests. I think we should opt to simply consume the fixture data directly instead of making a promise and having random errors.

Reply:
I can take out the random errors for sure but @hswolff thought it would be a better idea to have the fixtureData still resolve from a promise

Reply:
Yeah, I don't think it's particularly helpful to randomly error in people's applications in dev mode.

Reply:
😂

Reply:
That made me laugh, but removing it!